Configurable database names

dgoiko opened the pull request:
It is possible to configure the database names from the CrawlControler constructor.

It allowed me to create multiple CrawlerControllers on the same working directory without having the URLs mixed between them.

Substituting #420 because I broke the branch

Chaiavi commented:
This pull request makes sense to me, I think it should be merged

Please note that I didn't look deeply into the code, but this feature is a good one

dgoiko commented:
Thanks for your note. Just for the record, the only changes performed are changing constants which are used as arguments for sleepycat are now constructor arguments for the calling classes. I've added those constructors maintaining old ones for retro-compatibility. 

InProcessPagesDB was not configurable since I didn't need it, but now that someone showed interest for this I've just added it to the code

I was thinking about including it into CrawlConfig variable, however, I thought it would be harded to get merged (more code to review by maintainer).
